idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,659 @@
+
+            @self.Expression(self.flowsheet().config.time)
+            def coeff_b(b, t):
+                return 6.863 * exp(-0.01329 * b.Ang[t]) \
+                    + 0.02018 * exp(0.09443 * b.Ang[t])
+
+            @self.Expression(self.flowsheet().config.time)
+            def coeff_c(b, t):
+                return -0.00891 * exp(0.06757 * b.Ang[t]) \
+                    + 0.9661 * exp(0.006228 * b.Ang[t])
+        else:
+            self.coeff_a = Param(default=-428.64,
+                                 mutable=True,
+                                 doc="Performance Curve Parameter A")
+            self.coeff_b = Param(default=29.178,
+                                 mutable=True,
+                                 doc="Perforamance Curve Parameter B")
+            self.coeff_c = Param(default=0.9737,
+                                 mutable=True,
+                                 doc="Performance Curve parameter C")
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Correlation between psi_s and psi_3")
+        def psi_s_stage_eqn(b, t):
+            if self.config.first_stage is True:
+                return b.psi_s[t] == b.coeff_a[t] * b.psi_3[t]**2 + \
+                    b.coeff_b[t] * b.psi_3[t] + b.coeff_c[t]
+            else:
+                return b.psi_s[t] == b.coeff_a * b.psi_3[t]**2 + \
+                    b.coeff_b * b.psi_3[t] + b.coeff_c
+
+    def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
+                   optarg={'tol': 1e-6, 'max_iter': 50}):
+        """
+        Initialize the inlet compressor stage model.
+        This deactivates the specialized constraints,
+        then does the isentropic compressor initialization,
+        then reactivates the constraints and solves.
+
+        Args:
+            state_args (dict): Initial state for property initialization
+            outlvl (int): Amount of output (0 to 3) 0 is lowest
+            solver (str): Solver to use for initialization
+            optarg (dict): Solver arguments dictionary
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        flags = blk.control_volume.initialize(state_args=state_args,
+                                              hold_state=True)
+
+        opt = SolverFactory(solver)
+        opt.options = optarg
+
+        for t in blk.flowsheet().config.time:
+            # if there is not a good guess for efficiency or outlet pressure
+            # provide something reasonable.
+            eff = blk.efficiency_isentropic[t]
+            eff.fix(eff.value if value(eff) > 0.3 and value(eff
+                                                            ) < 1.0 else 0.85)
+            # check for alternate pressure specs
+            if blk.outlet.pressure[t].fixed:
+                blk.ratioP[t] = value(
```

Comment:
May want to throw the third pressure change variable in here (deltaP in this case).  There is a chance someone fixed two things to specify pressure change, and you may want to test for that, and raise and exception explaining it to the user.